components/server/src/workspace/workspace-starter.ts

```diff
@@ -332,6 +334,37 @@ export class WorkspaceStarter {
         }
     }
 
+    protected blockedRepositories: { urlRegExp: RegExp; blockUser: boolean }[] | undefined;
+    protected getBlockedRepositories() {
+        if (!this.blockedRepositories) {
```

Comment:
This should be checked at startup time, and move closer to our config.

There are two shapes in config.ts: Config and ConfigSerialized, and we actively translate between the two. This would be a perfect spot for this as well I think! This way we can also make sure regex parsing errors are caught and raised during "config loading time", and not later at runtime. :+1: